src/tokensource.ts

```diff
@@ -17,7 +17,7 @@
  */
 export class ManagedIdentityTokenSource {
     /** @hidden */
-    public kind: "AzureManagedIdentity";
+    public readonly kind: string = "AzureManagedIdentity";
```

Comment:
Hmm, was this field not being serialized before? If so, what about this change fixed that.

Reply:
It's a bit strange that TypeScript lets you do this, but `kind: "AzureManagedIdentity;"` is declaring "kind" as an undefined variable that will only be able to be assigned to the string "AzureManagedIdentity". So `kind === undefined` in this case

Reply:
This was my attempt at using [Discriminated Unions](https://www.typescriptlang.org/docs/handbook/advanced-types.html#discriminated-unions) in TypeScript, which Connor I think you recommended me to look into. I was surprised to see that the field was not included in the transpiled JavaScript at all.

Reply:
But yes, I verified end-to-end that this change fixes the issue.